modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2110,22 +2158,45 @@ boolean addTrailingCells(boolean fillEmptyCells) {
         return filledWithNonEmpty;
     }
 
-    void reconfigureCells() {
+    /**
+     * Informs the {@code VirtualFlow} that a layout pass should be done and cell contents (For example, in the {@code
+     * TableView}, if no cells has been added inside a row) are the same.
+     *
+     * @since 12
+     */
+    protected void reconfigureCells() {
         needsReconfigureCells = true;
         requestLayout();
     }
 
-    void recreateCells() {
+    /**
+     * Informs the {@code VirtualFlow} that a layout pass should be done and cell factory has changed. All cells are
+     * trashed and recreated to be replaced in the viewport.
+     *
```

Comment:
Same as above.

* comma before "and"
* "and cell factory" -> "and that a/the cell factory"
* What does "trashed" mean? Removed, deleted, dereferenced...?
* "recreated to be replaced" is odd. Maybe "recreated with the intent to be replaced"? I'm not sure what you're trying to say here.
* If we're talking about the cells in the viewport, then rearrange the position of "in the viewport" to "All cells in the viewport are trashed and recreated to be replaced".

Reply:
""recreated to be replaced" : The cell factory (aka the rowFactory for the TableView) has changed so every rows and cells are incorrect. So we must delete them all, and call the factory again to re-create new rows filled with new cells, then insert them in the viewport.

Reply:
OK, so I would write something like:
"... and that the cell factory has changed. All cells in the viewport are recreated with the new cell factory."